paperqa/docs.py

```diff
@@ -80,6 +80,11 @@ class Docs(BaseModel):
         default=PAPERQA_DIR, description="Path to save index", validate_default=True
     )
     deleted_dockeys: set[DocKey] = Field(default_factory=set)
+    partitioning_fn: Callable[[Doc], int] | None = Field(
```

Comment:
Won't this break serialization, so we need `exclude=True`

Or am I wrong here somehow?

Reply:
Yea it's prob safer to exclude, these should be pretty widely pickle-able, but if someone put a DB connection in their `partitioning_fn` then it wouldn't be. 

We could try to specify this with typing, but it's easier to just exclude.

Reply:
Down with whatever here